### PR TITLE
docs: update contributing guidelines

### PR DESCRIPTION
**Pull request by mpoke**
# Description

Update contributing guidelines to be consistent with other repos in the Cosmos stack, e.g., SDK, Gaia. 

## Linked issues

Closes: NA

## Type of change

If you've checked more than one of the first three boxes, consider splitting this PR into multiple PRs!

- [ ] `Feature`: Changes and/or adds code behavior, irrelevant to bug fixes
- [ ] `Fix`: Changes and/or adds code behavior, specifically to fix a bug
- [ ] `Refactor`: Changes existing code style, naming, structure, etc.
- [ ] `Testing`: Adds testing
- [x] `Docs`: Adds documentation

## Targeting

Please select one of the following:

- [x] This PR is only relevant to main
- [ ] This PR is relevant to main, and should also be back-ported to ____ (ex: v1.0.0 and v1.1.0)
- [ ] This PR is only relevant to ____ (ex: v1.0.0, v1.1.0, and v1.2.0)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,49 +1,179 @@
...
+    * Code must live in a reasonable location, considering dependency structures (for example, not importing testing modules in production code, or including example code modules in production code).
+    * If you approve the PR, you are responsible for any issues mentioned here and any issues that should have been addressed after thoroughly reviewing the reviewer checklist items in the pull request template.
+* If you sat down with the PR submitter and did a pairing review, add this information in the `Approval` or your PR comments.
+* If you are only making "surface level" reviews, submit notes as a `comment` review.
+
+### Updating Documentation
+
+If you open a PR in ICS, it is mandatory to update the relevant documentation in `/docs`.
+
+## Dependencies
```
Comment (mpoke) [inline CONTRIBUTING.md:149]: To be update after https://github.com/cosmos/interchain-security/issues/801 is completed. See https://github.com/cosmos/cosmos-sdk/blob/main/CONTRIBUTING.md#dependencies for an example.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,49 +1,179 @@
...
+
+For deterministic behavior around Protobuf tooling, everything is containerized using Docker. Make sure to have Docker installed on your machine, or head to [Docker's website](https://docs.docker.com/get-docker/) to install it.
+
+To generate the protobuf stubs, you can run `make proto-gen`.
+
+## Branching Model and Release
+
+ICS adheres to the [trunk based development branching model](https://trunkbaseddevelopment.com/). User branches should start with a user name, example: `{moniker}/{issue#}-branch-name`.
+
+### Semantic Versioning
```
Comment (mpoke) [inline CONTRIBUTING.md:176]: To be update after https://github.com/cosmos/interchain-security/issues/801 is completed. See https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#branching-model-and-release for an example.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,49 +1,179 @@
...
+* merge the latest main `git merge origin/main`,
+* run `make lint test` to ensure that all checks and tests pass.
+
+Then:
+
+1. If you have something to show, **start with a `Draft` PR**. It's good to have early validation of your work and we highly recommend this practice. A Draft PR also indicates to the community that the work is in progress.
+   Draft PRs also help the core team provide early feedback and ensure the work is in the right direction.
+2. When the code is complete, change your PR from `Draft` to `Ready for Review`.
+3. Go through the actions for each checkbox present in the PR template description. The PR actions are automatically provided for each new PR.
+4. Be sure to include a relevant changelog entry in the `Unreleased` section of `CHANGELOG.md` (see file for log format). The entry should be on top of all others changes in the section.
```
Comment (mpoke) [inline CONTRIBUTING.md:102]: Once https://github.com/cosmos/interchain-security/issues/801 is completed, we should start using [unclog](https://github.com/informalsystems/unclog).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,49 +1,179 @@
...
 
 ### ADR Proposals
 
 Architecture Decision Records (ADRs) may be proposed by any contributors or maintainers of ICS. ADRs are intended to be iterative, and may be merged into `main` while still in a `Proposed` status
 
-## PR Targeting
+## Development Procedure
+
+* The latest state of development is on `main`.
+* `main` must never fail `make lint test`.
```
Comment (mpoke) [inline CONTRIBUTING.md:71]: @smarshall-spitzbart please add a Makefile `lint` target

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,49 +1,179 @@
...
+* No `--force` onto `main` (except when reverting a broken commit, which should seldom happen).
+* Create a branch to start work:
+    * Fork the repo (core developers must create a branch directly in the ICS repo),
+    branch from the HEAD of `main`, make some commits, and submit a PR to `main`.
+    * For core developers working within the `interchain-security` repo, follow branch name conventions to ensure clear
+    ownership of branches: `{moniker}/{issue#}-branch-name`.
+    * See [Branching Model](#branching-model-and-release) for more details.
+* Be sure to run `make format` before every commit. The easiest way
```
Comment (mpoke) [inline CONTRIBUTING.md:79]: @smarshall-spitzbart please add a `format` target to the Makefile

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,49 +1,179 @@
...
-## Semantic Versioning
+When dependencies in ICS's `go.mod` are changed, it is generally accepted practice
+to delete `go.sum` and then run `go mod tidy`.
+
+Since some dependencies are not under our control, a third party may break our
+build, in which case we can fall back on `go mod tidy -v`.
+
+## Protobuf
+
+We use [Protocol Buffers](https://developers.google.com/protocol-buffers) along with [gogoproto](https://github.com/cosmos/gogoproto) to generate code for use in ICS.
```
Comment (MSalopek) [inline CONTRIBUTING.md:166]:
```suggestion
We use [Protocol Buffers](https://developers.google.com/protocol-buffers) along with [buf](https://buf.build/docs/) to generate code for use in ICS.
```

We use `buf generate` to generate code.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -1,43 +1,49 @@
...
-
-## Targeting
-
-Please select one of the following:
-
-- [ ] This PR is only relevant to main
-- [ ] This PR is relevant to main, and should also be back-ported to ____ (ex: v1.0.0 and v1.1.0)
-- [ ] This PR is only relevant to ____ (ex: v1.0.0, v1.1.0, and v1.2.0)
+* [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
+* [ ] confirmed `!` in the type prefix if API or client breaking change
```
[the remark this answers is not in the document]
Comment (mpoke) [reply .github/PULL_REQUEST_TEMPLATE.md:44]: ditto

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -1,43 +1,49 @@
-# Description
+<!--
```
[the remark this answers is not in the document]
Comment (mpoke) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: > Doesn't github allow you to choose templates when you try to create a PR?

Not sure. Let me know if you know of any. This is the approach from the SDK repo.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -1,43 +1,49 @@
...
-- [ ] `Testing`: Adds testing
-- [ ] `Docs`: Adds documentation
+*All items are required. Please add a note to the item if the item is not applicable and
+please add links to any relevant follow up issues.*
 
-## Regression tests
+I have...
 
-If `Refactor`, describe the new or existing tests that verify no behavior was changed or added where refactors were introduced.
+* [ ] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
```
[the remark this answers is not in the document]
Comment (mpoke) [reply .github/PULL_REQUEST_TEMPLATE.md:24]: I find the info at [the provided link](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) clear enough.

File: CONTRIBUTING.md (hunk 5 above)
Comment (smarshall-spitzbart) [reply CONTRIBUTING.md:79]: See https://github.com/cosmos/interchain-security/pull/860, it's similar to gaia but not completely the same

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (smarshall-spitzbart) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: See https://stackoverflow.com/questions/73771068/multiple-templates-for-pull-requests-on-github, we can make the default template link to the different template options

File: .github/PULL_REQUEST_TEMPLATE/docs.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+*All items are required. Please add a note if the item is not applicable and please add
+your handle next to the items reviewed if you only reviewed selected items.*
+
+I have...
+
+- [ ] Confirmed the correct `docs:` prefix in the PR title
+- [ ] Confirmed all author checklist items have been addressed 
+- [ ] Confirmed that this PR only changes documentation
+- [ ] Reviewed content for consistency
```
Comment (smarshall-spitzbart) [inline .github/PULL_REQUEST_TEMPLATE/docs.md:33]: Do you mean formatting consistency?
```suggestion
- [ ] Reviewed content for formatting consistency
```

Comment (mpoke) [reply .github/PULL_REQUEST_TEMPLATE/docs.md:33]: No. Content consistency. Changes throughout the docs should be consistent.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (mpoke) [reply .github/PULL_REQUEST_TEMPLATE.md:44]: Yes, but it doesn't hit to emphasis this point as it's important.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 11]
```diff
@@ -1,43 +1,49 @@
...
+* [ ] Added `!` to the type prefix if API or client breaking change
+* [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/interchain-security/blob/main/CONTRIBUTING.md#pr-targeting))
```
[the remark this answers is not in the document]
Comment (mpoke) [reply .github/PULL_REQUEST_TEMPLATE.md:26]: We should enable mergify (similar to Gaia) once we split go.mod. Those PRs will be automatically open and the info on how to do that will be in a `RELEASE_PROCESS.md` doc. See https://github.com/cosmos/gaia/pull/2290 for an example.

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -1,49 +1,179 @@
...
+We use [Go Modules](https://github.com/golang/go/wiki/Modules) to manage
+dependency versions.
 
-- The latest state of development is on `main`
-- `main` must never fail `make test`
-- Create a branch or fork the repo to start work
-- Once development is completed, or during development as a draft, you may open a pull request. The pull request description should follow the [default template](./.github/PULL_REQUEST_TEMPLATE.md). Note that a PR can be of one or multiple types, including `Feature`, `Fix`, `Refactor`, `Testing`, and `Docs`. Please try to minimize the number of categories that you could classify a single PR as.
-- It's notable that your PR may have versioning implications. See [Semantic Versioning](#semantic-versioning) for more information
-- Code is squash merged through the PR approval process
+The main branch of every Cosmos repository should just build with `go get`,
```
[the remark this answers is not in the document]
Comment (mpoke) [reply CONTRIBUTING.md:152]: Just got the entire section from the SDK guidelines.

**Review by MSalopek — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (mpoke) [reply CONTRIBUTING.md:148]: cc @smarshall-spitzbart

File: CONTRIBUTING.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (mpoke) [reply CONTRIBUTING.md:153]: See https://github.com/cosmos/interchain-security/pull/859#discussion_r1169917068

**Review by MSalopek — APPROVED** [review]
Thanks for including the requested changes!

**Review by sainoe — APPROVED** [review]

**Review by jtremback — COMMENTED** [review]
These seem sane

**Pull request merged**